packages/router/src/std/routes/connect.ts

```diff
@@ -0,0 +1,40 @@
+import { localExtStorage } from '@penumbra-zone/storage';
+import {
+  ConnectMessage,
+  MessageStatus,
+  MessageType,
+  ServicesInterface,
+  SwMessageHandler,
+} from '@penumbra-zone/types';
+import generateUniqueId from 'generate-unique-id';
+
+export const connectHandler =
+  (
+    services: ServicesInterface,
+    sender: chrome.runtime.MessageSender,
+  ): SwMessageHandler<ConnectMessage> =>
+  async () => {
+    const connectedSites = await localExtStorage.get('connectedSites');
+    const messages = await localExtStorage.get('messages');
+
+    // check is pending message exist with same origin
+    if (messages.find(msg => msg.status === MessageStatus.PENDING && msg.origin === sender.origin))
```

Comment:
What is a message and what does it mean for a message to be pending?

Reply:
when user click connect, we have message with pending status (need to show active message page), after user can Approve or Reject message